contracts/cw-token-swap/src/msg.rs

```diff
@@ -0,0 +1,60 @@
+use cosmwasm_std::Uint128;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use crate::state::CheckedCounterparty;
+
+/// Information about the token being used on one side of the escrow.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, JsonSchema)]
+#[serde(rename_all = "snake_case")]
+pub enum TokenInfo {
+    /// A native token.
+    Native { denom: String, amount: Uint128 },
```

Comment:
Also isn't this just a duplicate of the same code in `state.rs` why not use that here

Reply:
One is a checked and one is not. The checked edition uses `Addr` instead of `String`. Pretty similar, but I like separating them like this. :)